CONTRIBUTING.md

```diff
@@ -30,6 +30,22 @@ works.
 
 This document will guide you through the contribution process.
 
+### Rules of thumb
+
+1. __Provide motivation for the change__  
+   Why will this change make the code better; does it fix a but, is it a new
+   feature, etc.
+2. __Don't cause _unnecessary_ code churn__  
+   As part of the projects strategy we maintain several lines of code, code 
+   churn might hinder back-porting changes to other branches. Also when you 
+   change a line, your name will come up in `git blame` and might hide the
+   previous writer of the code.
+3. __Keep your change-set self contained but at a reasonable size__  
+   30 files is a reasonable size limit, so unless you have a good reason to make
+   a bigger PR, try to break it into smaller pieces (still as self-contained
+   as possible), and cross-reference them. You can also mark some of them as
+   `blocked` pending the others.   
```

Comment:
I don’t think we should set up rules like this one, and instead keep trying to use good judgement. I can’t really imagine a “reasonable size” defined for all PRs, it’s really a case-by-case thing. (e.g. #11883 and #11975 are huge PRs and that’s completely reasonable in my eyes).